sdk/javascript/src/symbol/BlockFactory.js

```diff
@@ -0,0 +1,91 @@
+import { Address } from './Network.js';
+import * as sc from './models.js';
+import ByteArray from '../ByteArray.js';
+import { Hash256, PublicKey } from '../CryptoTypes.js';
+import RuleBasedTransactionFactory from '../RuleBasedTransactionFactory.js';
+
+/**
+ * Represents a 128-bit proof verification hash.
+ */
+class ProofVerificationHash extends ByteArray {
+	static SIZE = 16;
+
+	/**
+	 * Creates a proof verification hash from bytes or a hex string.
+	 * @param {Uint8Array|string} proofVerificationHash Input string or byte array.
+	 */
+	constructor(proofVerificationHash) {
+		super(ProofVerificationHash.SIZE, proofVerificationHash);
+	}
+}
+
+/**
+ * Factory for creating Symbol blocks.
+ */
+export default class BlockFactory {
+	/**
+	 * Creates a factory for the specified network.
+	 * @param {Network} network Symbol network.
+	 * @param {Map} typeRuleOverrides Type rule overrides.
+	 */
+	constructor(network, typeRuleOverrides) {
+		this.factory = BlockFactory.buildRules(typeRuleOverrides);
+		this.network = network;
+	}
+
+	_createAndExtend(blockDescriptor, autosort, FactoryClass) {
+		const block = this.factory.createFromFactory(FactoryClass.createByName, {
+			...blockDescriptor,
+			network: this.network.identifier
+		});
+		if (autosort)
+			block.sort();
+
+		return block;
+	}
+
+	/**
+	 * Creates a block from a block descriptor.
+	 * @param {object} blockDescriptor Block descriptor.
+	 * @param {boolean} autosort When set (default), descriptor arrays requiring ordering will be automatically sorted.
+	 *                           When unset, descriptor arrays will be presumed to be already sorted.
+	 * @returns {object} Newly created transaction.
+	 */
+	create(blockDescriptor, autosort = true) {
+		return this._createAndExtend(blockDescriptor, autosort, sc.BlockFactory);
+	}
+
+	static _symbolTypeConverter(value) {
+		if (value instanceof Address)
+			return new sc.UnresolvedAddress(value.bytes);
+
+		return undefined;
+	}
+
+	static buildRules(typeRuleOverrides) {
+		const factory = new RuleBasedTransactionFactory(sc, this._symbolTypeConverter, typeRuleOverrides);
+		factory.autodetect();
+
+		// block.network (networkId)
+		['NetworkType'].forEach(name => { factory.addEnumParser(name); });
+
+		factory.addStructParser('VrfProof');
+
+		const sdkTypeMapping = {
+			UnresolvedAddress: Address,
+			Address,
+			Hash256,
+			PublicKey,
+			ProofGamma: Hash256, // should introduce type?
+			ProofVerificationHash,
+			ProofScalar: Hash256 // should introduce type?
+		};
+		Object.keys(sdkTypeMapping).forEach(name => { factory.addPodParser(name, sdkTypeMapping[name]); });
+
+		['BlockType', 'UnresolvedAddress'].forEach(name => {
```

Comment:
shouldn't BlockType be enum?

Reply:
BlockType _is_ an enum, this adds enum-array parser...